nevergrad/parametrization/core.py

```diff
@@ -0,0 +1,497 @@
+        """Random state the instrumentation and the optimizers pull from.
+        It can be seeded/replaced.
+        """
+        if self._random_state is None:
+            # use the setter, to make sure the random state is propagated to the variables
+            seed = np.random.randint(2 ** 32, dtype=np.uint32)
+            self._set_random_state(np.random.RandomState(seed))
+        assert self._random_state is not None
+        return self._random_state
+
+    @random_state.setter
+    def random_state(self, random_state: np.random.RandomState) -> None:
+        self._set_random_state(random_state)
+
+    def _set_random_state(self, random_state: np.random.RandomState) -> None:
+        self._random_state = random_state
+        if self._subparameters is not None:
+            self.subparameters._set_random_state(random_state)
+
+    def spawn_child(self: P) -> P:
+        rng = self.random_state  # make sure to create one before spawning
+        child = self._internal_spawn_child()
+        child._set_random_state(rng)
+        child._constraint_checkers = list(self._constraint_checkers)
+        child._generation = self.generation + 1
+        child.parents_uids.append(self.uid)
+        return child
+
+    def _internal_spawn_child(self: P) -> P:
+        # default implem just forwards params
+        inputs = {k: v.spawn_child() if isinstance(v, Parameter) else v for k, v in self.subparameters._parameters.items()}
+        child = self.__class__(**inputs)
+        return child
+
+
+class Constant(Parameter):
+    """Parameter-like object for to simplify management of constant parameters:
+    mutation/recombination do nothing, value canot be changed, standardize data is an empty array,
+    child is the same instance.
+
+    Parameter
+    ---------
+    value: Any
+        the value that this parameter will always provide
+    """
+
+    def __init__(self, value: t.Any) -> None:
+        super().__init__()
+        if isinstance(value, Parameter):
+            raise TypeError("Only non-parameters can be wrapped in a Constant")
+        self._value = value
+
+    def _get_name(self) -> str:
+        return str(self._value)
+
+    @property
+    def value(self) -> t.Any:
+        return self._value
+
+    @value.setter
+    def value(self, value: t.Any) -> None:
+        if not value == self._value:
+            raise ValueError(f'Constant value can only be updated to the same value (in this case "{self._value}")')
+
+    def _internal_get_std_data(self: BP, instance: BP) -> np.ndarray:
+        return np.array([])
+
+    def _internal_set_std_data(self: P, data: np.ndarray, instance: P, deterministic: bool = False) -> P:
+        if data.size:
+            raise ValueError(f"Constant dimension should be 0 (got data: {data})")
+        return instance
+
+    def spawn_child(self: P) -> P:
+        return self  # no need to create another instance for a constant
+
+
+def as_parameter(param: t.Any) -> Parameter:
```

Comment:
Nit: Maybe easier to put that in the `Parameter` class directly? `Parameter.from(x)`

Reply:
I thought of something like this but if it's in `Parameter` then all parameters would inherit from it, while this is not to be used outside of the implementations (it would not make sense). I don't like much the current way of doing it either though.
Another option was to convert **everything** as parameter, but I thought that would be messier. Not sure though...